hcl2template/types.packer_config.go

```diff
@@ -550,6 +554,22 @@ func (cfg *PackerConfig) GetBuilds(opts packer.GetBuildsOptions) ([]packersdk.Bu
 			res = append(res, pcb)
 		}
 	}
+	if len(opts.Only) > opts.OnlyMatches {
+		diags = append(diags, &hcl.Diagnostic{
+			Severity: hcl.DiagWarning,
+			Summary:  "an 'only' option was passed, but did not match any build.",
```

Comment:
A bit of a weird edge case as a user would have to specify more `only` opts than actual defined sources. But I think it could be a bit misleading to the user if they accidentally fall into the edge case so I'm suggesting a small text change. See use case below. 
 
```suggestion
			Summary:  "an 'only' option was passed, but did not all matches were found for the given build.",
```


```bash
~>  packer build -debug -only=upcloud.test,amazon-ebs.example .
Error: 2 error(s) occurred:

* 'username' must be specified
* 'password' must be specified

  on source.pkr.hcl line 20:
  (source code not available)

Warning: an 'only' option was passed, but did not match any build.

Possible build names: [upcloud.test upcloud.example].
These could also be matched with a glob pattern like: 'happycloud.*'


Debug mode enabled. Builds will not be parallelized.

==> Wait completed after 10 microseconds

==> Builds finished but no artifacts were created.
```

Reply:
Oh yeah that's a good idea !